custom_components/smartir/light.py

```diff
@@ -150,155 +127,189 @@ def extra_state_attributes(self):
-                        steps = abs(steps)
-                    else:
-                        cmd = CMD_BRIGHTNESS_INCREASE
+    async def async_turn_on(self, **kwargs):
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        color_temp = kwargs.get(ATTR_COLOR_TEMP_KELVIN)
 
-                    if steps > 0 and cmd:
-                        # If we are heading for the highest or lowest value,
-                        # take the opportunity to resync by issuing enough
-                        # commands to go the full range.
-                        if (
-                            new_brightness == len(self._brightnesses) - 1
-                            or new_brightness == 0
-                        ):
-                            steps = len(self._brightnesses)
-                        self._brightness = self._brightnesses[new_brightness]
-                        await self.send_command(cmd, steps)
-
-        # If we did nothing above, and the light is not detected as on
-        # already issue the on command, even though we think the light
-        # is on.  This is because we may be out of sync due to use of the
-        # remote when we don't have anything to detect it.
-        # If we do have such monitoring, avoid issuing the command in case
-        # on and off are the same remote code.
-        if not did_something and not self._on_by_remote:
-            self._state = STATE_ON
-            await self.send_command(CMD_POWER_ON)
-
-        self.async_write_ha_state()
+        if self._brightness_list is not None and brightness is None:
+            _LOGGER.debug(
+                "No power on brightness argument found, setting last brightness '%s'",
+                self._brightness,
+            )
+            brightness = self._brightness
+
+        if self._color_temp_list is not None and color_temp is None:
+            _LOGGER.debug(
+                "No power on color temperature argument found, setting last color temperature '%s'",
+                self._color_temp,
+            )
+            color_temp = self._color_temp
+
+        await self._send_command(STATE_ON, brightness, color_temp)
 
     async def async_turn_off(self):
-        if self._state != STATE_OFF:
-            self._state = STATE_OFF
-            await self.send_command(CMD_POWER_OFF)
-            self.async_write_ha_state()
+        await self._send_command(STATE_OFF)
 
     async def async_toggle(self):
         await (self.async_turn_on() if not self.is_on else self.async_turn_off())
 
-    async def send_command(self, cmd, count=1):
-        if cmd not in self._commands:
-            _LOGGER.error(f"Unknown command '{cmd}'")
-            return
-        _LOGGER.debug(f"Sending {cmd} remote command {count} times.")
-        remote_cmd = self._commands.get(cmd)
-        await self.send_remote_command(remote_cmd, count)
-
-    async def send_remote_command(self, remote_cmd, count=1):
+    async def send_command(self, state, brightness, color_temp):
```

Comment:
`_send_command` (prefixed by underscore) to be consistent with other uses in this file.

brightness and color_temp should have default values so the call with one argument from async_turn_off works.

Reply:
I did unify send_command as it was all over the places. Also we can set brightness and colorTemperature to anything as in STATE_OFF it is not sent. Now fixed as well.